### PR TITLE
Add support for MessageReaction#remove and MESSAGE_REACTION_REMOVE_EMOJI

### PR DESCRIPTION
**Pull request by RDambrosio016**
**Please describe the changes this PR makes and why it should be merged:**
This pr adds support for the new event MESSAGE_REACTION_REMOVE_EMOJI documented in https://github.com/discordapp/discord-api-docs/pull/1309, it adds a new method to `MessageReaction` to remove it (remove all occurences of an emoji reaction from a message). it adds another event which is triggered when a bot does this too.

**Status**
- [x] Code changes have been tested against the Discord API, or there are no code changes
- [x] I know how to update typings and have done so, or typings don't need updating

**Semantic versioning classification:**  
- [x] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: src/client/actions/MessageReactionRemoveEmoji.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+class MessageReactionRemoveEmoji extends Action {
+  handle(data) {
+    const channel = this.getChannel(data);
+    if (!channel || channel.type === 'voice') return false;
+
+    const message = this.getMessage(data, channel);
+    if (!message) return false;
+
+    const reaction = this.getReaction(data, message);
+    if (!reaction) return false;
```
Comment (matthewfripp) [inline src/client/actions/MessageReactionRemoveEmoji.js]:
```suggestion
    if (!reaction) return false;
    message.reactions.delete(reaction.emoji.id || reaction.emoji.name);
```
You should delete the reaction from cache

Comment (RDambrosio016) [reply src/client/actions/MessageReactionRemoveEmoji.js]: 👍

File: src/client/actions/MessageReactionRemoveEmoji.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    if (!message.partial) message.reactions.delete(reaction.emoji.id || reaction.emoji.name);
+
+    /**
+     * Emitted when a bot removes a reaction from a cached message.
```
Comment (amishshah) [inline src/client/actions/MessageReactionRemoveEmoji.js]: It does more than remove a reaction right? It should detail that it removes all reactions of a particular emoji from a message.

Comment (RDambrosio016) [reply src/client/actions/MessageReactionRemoveEmoji.js]: Well, a reaction is known to both the api and the lib as containing a single emoji and multiple users, so its technically removing a single reaction, i am uncertain of the wording i should use for exactly this reason

Comment (tipakA) [reply src/client/actions/MessageReactionRemoveEmoji.js]: How can you differentiate singular person clicking on their reaction to remove it, from someone removing whole reaction then? Both are reactions removals, except the latter wasn't possible before.
Personally, i think i would use word `emoji` there, afterall you remove emoji, no matter how many reactions it had.

Also, remember that while yes, for API and library it's reaction, it not always has to be instantly logical for actual users. Just think how many times you have to explain difference between User and GuildMember.

**Comment by RDambrosio016** [comment]
> `MessageReaction#remove()` is missing from typings

Fixed in `ebefc90`

**Pull request merged**